pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -351,23 +389,11 @@ def __process_eval_epoch_end_results_and_log_legacy(self, eval_results, test_mod
                 else:
                     _, prog_bar_metrics, log_metrics, callback_metrics, _ = self.trainer.process_dict_result(result)
 
-                # eval loop returns all metrics
-                dataloader_result_metrics = {**prog_bar_metrics, **log_metrics, **callback_metrics}
-
-                # add metrics to prog bar
-                self.trainer.logger_connector.add_progress_bar_metrics(prog_bar_metrics)
-
-                # log metrics
-                if len(log_metrics) > 0:
-                    self.trainer.logger_connector.log_metrics(log_metrics, {})
-
-                # track metrics for callbacks (all prog bar, logged and callback metrics)
-                self.trainer.logger_connector.callback_metrics.update(callback_metrics)
-                self.trainer.logger_connector.callback_metrics.update(log_metrics)
-                self.trainer.logger_connector.callback_metrics.update(prog_bar_metrics)
+                if num_loaders > 1:
+                    self.__process_eval_epoch_end_results_and_log_legacy_udpate(prog_bar_metrics, log_metrics, callback_metrics)
 
-                if len(dataloader_result_metrics) > 0:
-                    self.eval_loop_results.append(dataloader_result_metrics)
+            if num_loaders == 1:
+                self.__process_eval_epoch_end_results_and_log_legacy_udpate(prog_bar_metrics, log_metrics, callback_metrics)
```

Comment:
PyRight is throwing an error here as these variables can be unbound

```  
396:77 - error: "prog_bar_metrics" is possibly unbound (reportUnboundVariable)
396:95 - error: "log_metrics" is possibly unbound (reportUnboundVariable)
396:108 - error: "callback_metrics" is possibly unbound (reportUnboundVariable)
```

Reply:
Done !